commit-check

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+set -e
+
+
+function cleanup()
+{
+    echo
+    echo "Cleaning up..."
+    if [[ -d $TMP_VENV ]]; then
+        rm -r "$TMP_VENV"
+    fi
+}
+
+trap cleanup EXIT
+
+
+#=====================================================
+# Run checks
+#=====================================================
+
+FAILURES=0
+
+echo "Setting up python venv..."
+TMP_VENV=$(mktemp -d -t venv.XXXXXX)
```

Comment:
Do we definitely want to be force using a temporary venv rather than giving the option of a users own? Is this just to keep it nice and simple and if a user was advanced enough they'd install their own venv and run each tool themselves?

Reply:
Yeah wanted to minimise complexity (no arg parsing). As per the contributing guide, this is just a fire-and-forget convenience, where the expectation is for devs to have their own venv set up and run tools individually in most cases.

Reply:
This is partly serving to illustrate the commands that should be run when setting things up manually.